CenterForOpenScience/ember-osf-web
# [EMB-537] osf-link

## Purpose

To ensure we are using one type of Link in the main app, and to track the clicks with those links and buttons.

## Summary of Changes
* Rename `Link` component to `OsfLink` and standardize the interface just a smidge
* Replace nearly all uses of other linking options in main app to `OsfLink`
* Add `data-analytics-context` identifiers to all pages and some components
* Add `data-analytics-name` identifiers to all clickable items we control in main app
* Use new `click` test helper that verifies `data-analytics-name` existence
* Remove old analytics tracking
* Update handbook and tests

## QA Notes

Links that go to places not within Ember-osf-web are untestable, although there are fewer ways to mess that up than there were before. You can see the minor visual differences at the Percy section of the Checks below, or go to 
https://percy.io/Center-for-Open-Science/ember-osf-web/builds/1325054

There's a list of thing below that aren't well covered by tests, so hypothetically there could be a problem. I'm reasonably certain that testing one will test all of them.

## Ticket


https://openscience.atlassian.net/browse/EMB-537

# Reviewer Checklist




# Not well covered by tests (hrefs)
## Dashboard

- data-analytics-name='noteworthy_search'
- data-analytics-name='meetings_button'
- data-analytics-name='preprints_button'
- Project links in user’s projects
- project links in New and Popular lists
- Institutions
- Create project - Go to new

## Home

- Youtube video
- Explore activity
- Cos.io
- #signup
- Institutions
- institutions read more
- Visit institution
- Contact us

## Sign up page

- ORCid
- Institution

## Settings side nav bar

- addons
- notifications

## Support
- All the things

## Contributors list

- Click on a contributor
- And x more

## File share button

- Twitter
- facebook
- linkedin
- email

## Join OSF Banner

- sign-up
- learn more

## Node card (forks page) dropdown menu

- Contributors
- Settings

## Node and registrations analytics pages navbar

- Settings

## Footer
- All the things

## Nav bar user dropdown

- Sign up
- My profile
- Settings

## Nav bar

- My projects
- Preprints
